File: iterableapi/src/main/java/com/iterable/iterableapi/IterablePushRegistration.java
package com.iterable.iterableapi;

import java.util.concurrent.Executor;

final class IterablePushRegistration {
    private final Executor executor;

    IterablePushRegistration() {
        this(IterableExecutors.push());
    }

    IterablePushRegistration(Executor executor) {
        this.executor = executor;
    }

    void executePushRegistrationTask(IterablePushRegistrationData data) {
        executor.execute(new IterablePushRegistrationTask(data));
    }
}

File: iterableapi/src/main/java/com/iterable/iterableapi/IterablePushRegistration.kt
package com.iterable.iterableapi

import androidx.annotation.VisibleForTesting

internal object IterablePushRegistration {

    @VisibleForTesting
    var instance = IterablePushRegistrationImpl()

    fun executePushRegistrationTask(data: IterablePushRegistrationData) {
        instance.executePushRegistrationTask(data)
    }

    class IterablePushRegistrationImpl {
        fun executePushRegistrationTask(data: IterablePushRegistrationData) {
            IterablePushRegistrationTask().execute(data)
        }
    }
}
